filebeat/registrar/registrar.go

```diff
@@ -83,23 +83,23 @@ func (r *Registrar) loadStates() error {
 		return nil
 	}
 
-	file, err := os.Open(r.registryFile)
+	f, err := os.Open(r.registryFile)
 	if err != nil {
 		return err
 	}
 
-	defer file.Close()
+	defer f.Close()
 
 	logp.Info("Loading registrar data from %s", r.registryFile)
 
 	// DEPRECATED: This should be removed in 6.0
-	oldStates := r.loadAndConvertOldState(file)
+	oldStates := r.loadAndConvertOldState(f)
 	if oldStates {
 		return nil
 	}
 
-	decoder := json.NewDecoder(file)
-	states := []input.FileState{}
+	decoder := json.NewDecoder(f)
+	states := []file.State{}
 	decoder.Decode(&states)
```

Comment:
add missing error check here. What's the content of states if registry has been corrupted?